src/Aspire.Hosting.Azure/Provisioning/Provisioners/AzureProvisioner.cs

```diff
@@ -362,14 +364,44 @@ await foreach (var tenant in armClient.GetTenants().GetAllAsync(cancellationToke
             throw new MissingConfigurationException("An azure location/region is required. Set the Azure:Location configuration value.");
         }
 
-        var unique = $"{Environment.MachineName.ToLowerInvariant()}-{environment.ApplicationName.ToLowerInvariant()}";
+        var userSecrets = await userSecretsLazy.Value.ConfigureAwait(false);
+
+        string resourceGroupName;
+        bool createIfAbsent;
 
-        // Name of the resource group to create based on the machine name and application name
-        var (resourceGroupName, createIfAbsent) = _options.ResourceGroup switch
+        if (string.IsNullOrEmpty(_options.ResourceGroup))
         {
-            null or { Length: 0 } => ($"rg-aspire-{unique}", true),
-            string rg => (rg, _options.AllowResourceGroupCreation ?? false)
-        };
+            // Generate an resource group name since none was provided
+
+            var prefix = "rg-aspire";
+
+            if (!string.IsNullOrWhiteSpace(_options.ResourceGroupPrefix))
+            {
+                prefix = _options.ResourceGroupPrefix;
```

Comment:
Why aren't we normalizing the `ResourceGroupPrefix`?

Reply:
For the same reason we are not doing it for the `ResourceGroup` option? I think no since this is a technical value provided by the user. For `ApplicationName` the user should not have to worry about naming limitations.

Otherwise let's do it for both. And maybe log a warning that the name was changed if provided?

Reply:
Ok, if we think that since this option is named "ResourceGroupPrefix", users will be responsible for using the correct characters.